src/xml.h

```diff
@@ -81,6 +81,7 @@ struct lyxml_ctx {
 
     enum LYXML_PARSER_STATUS status; /* status providing information about the last parsed object, following attributes
                                         are filled based on it */
+    uint32_t depth;         /* current number of nested blocks, see ::LY_MAX_BLOCK_DEPTH */
```

Comment:
I am a bit unsure as to whether to add this explicit variable or not. But since you can directly use `xmlctx->elements.count` since the elements are always [removed](https://github.com/CESNET/libyang/blob/libyang2/src/xml.c#L547) and [added](https://github.com/CESNET/libyang/blob/libyang2/src/xml.c#L601), I think it is redundant. Comparing directly on the open element count has also the advantage of being obvious that it is actually limit on the open elements since there are no blocks in XML.